braindecode/losses.py

```diff
@@ -1,10 +1,14 @@
 import torch
 
 
-class CroppedNLLLoss:
-    """Compute NLL Loss after averaging predictions across time.
+class CroppedLoss:
+    """Compute Loss after averaging predictions across time.
     Assumes predictions are in shape:
     n_batch size x n_classes x n_predictions (in time)"""
 
+    def __init__(self, loss_function):
+        self.loss_function = loss_function
+
     def __call__(self, preds, targets):
-        return torch.nn.functional.nll_loss(torch.mean(preds, dim=2), targets)
+        avg_preds = torch.mean(preds, dim=2).squeeze()
```

Comment:
what is purpose of squeeze here? torch squeeze by default anyways dim 2 right? and if you have batch size 1 this will lead to error also right? I think in general squeeze should always be called with dimension argument to avoid errors in any case

Reply:
So far, code was written to perform multi-class classification. Predictions here are given as <code>batch_size x n_classes</code>. For regression, <code>n_classes = 1</code> and I get a UserWarning of incompatible sahpes when computing <code>mse_loss</code>. Therefore, I want to squeeze empty first dimension. Is there a better way to do this?

Reply:
How about solving this issue in a way that multi-target regression works too? In my case it may be useful as I have to predict x, y, z coordinates. Do you think it is possible?

Reply:
If I understand correctly, in your setting predictions would be in shape <code>batch_size x 3 x n_times</code> and targets in shape <code>batch_size x 3</code>. For <code>mse_loss</code> predictions and targets should have the same dimensionality. So this should work with <code>squeeze(dim=1)</code>, right?

Reply:
@gemeinl yeah, it should work. I checked and in this setup `torch.mean` already returns tensor of shape `batch_size x 3`, so the `squeeze` operation is not needed. Is it added for other cases?

Reply:
Yes, we only need squeezing when <code>n_classes=1</code>.

Reply:
yeah so for sure add dim to squeeze. then should be fine and extensible i guess